### PR TITLE
move online vs offline explanation to better location. move upgrade to better location

### PR DESCRIPTION
**Pull request by mattsfuller**

### CODE REVIEW

File: docs/installation/presto-admin-installation.rst [hunk 1]
```diff
@@ -5,19 +5,42 @@ Installing Presto Admin
 =======================
 *Prerequisites:* `Python 2.6 or Python 2.7 <https://www.python.org/downloads>`_
 
+To install ``presto-admin``:
```
Comment (rschlussel) [inline docs/installation/presto-admin-installation.rst]: move this down to above the list of instructions.

File: docs/installation/presto-admin-installation.rst [hunk 2]
```diff
@@ -5,19 +5,42 @@ Installing Presto Admin
...
+offline installer is currently only supported on RedHat Linux 6.x or
+CentOS equivalent.
+
+The online installer downloads all of the dependencies when you call
+``./install-prestoadmin.sh``. You need to use the online installer
+for installation of Presto on Amazon EMR.
+
+You may also need to use the online installer if you try to install
+``presto-admin`` on an operating system other than listed above,
+because some of the binary files are dependent on the system files of
```
Comment (rschlussel) [inline docs/installation/presto-admin-installation.rst]: this seems like more information than a user needs to know

File: docs/installation/presto-admin-installation.rst [hunk 3]
```diff
@@ -5,19 +5,42 @@ Installing Presto Admin
...
+Presto Admin is packaged as an offline installer --
+``prestoadmin-<version>-offline.tar.bz2`` -- and as an online
+installer -- ``prestoadmin-<version>-online.tar.bz2``. The offline
+installer includes all of the dependencies for ``presto-admin``, so it
+can be used on a cluster without an outside network connection. The
+offline installer is currently only supported on RedHat Linux 6.x or
+CentOS equivalent.
+
+The online installer downloads all of the dependencies when you call
+``./install-prestoadmin.sh``. You need to use the online installer
```
Comment (rschlussel) [inline docs/installation/presto-admin-installation.rst]: Maybe combine this with the paragraph below:

You must use the online installer for installation of Presto on Amazon EMR and for use on any operating system not listed above.  If you are using ``presto-admin`` on an unsupported operating system, beware that there may be operating system dependencies beyond the installation process, and ``presto-admin`` may not work.

File: docs/installation/presto-admin-installation.rst [hunk 4]
```diff
@@ -5,19 +5,42 @@ Installing Presto Admin
...
+installation location is ``/opt``. Note that ``presto-admin`` does not
+have to be on the same node(s) where Presto will run, though it does
+need to have SSH access to all of the nodes in the cluster.
+
 .. NOTE::
-     For Amazon EMR, use the online installer instead of the offline installer (see :ref:`advanced-installation-options-label`).
+     For Amazon EMR, use the online installer instead of the offline installer.
 
 .. NOTE::
    Omit the use of ‘sudo’ when the prestoadmin scripts and presto-admin commands when the root user is selected as the prestoadmin user.
```
Comment (rschlussel) [inline docs/installation/presto-admin-installation.rst]: This note seems unnecessary since nothing bad happens if you use sudo while already the root user.  Could you delete it since you're here?

File: docs/installation/presto-admin-installation.rst [hunk 5]
```diff
@@ -5,19 +5,42 @@ Installing Presto Admin
...
 
 .. NOTE::
    The sudo setup for a non-root user has a security hole where the presto-admin non-root user requires the ability to run /bin/bash as root. The IT organization will need to take the appropriate steps to address this security hole and select which user will be the presto-admin user.
```
Comment (rschlussel) [inline docs/installation/presto-admin-installation.rst]: I found this sentence a bit hard to follow.  Maybe change it as below:

The sudo setup for a non-root user must have the ability to run `/bin/bash` as root.  This can be a security issue.  The IT organization should take the necessary steps to address this security hole and select an appropriate `presto-admin` user.

Comment (rschlussel) [inline docs/installation/presto-admin-installation.rst]: also this is for the presto-admin user configured in config.json.  This note should be in the docs there.

File: docs/installation/presto-admin-installation.rst (hunk 2 above)
Comment (mattsfuller) [inline docs/installation/presto-admin-installation.rst]: right. I simply copied this from the other location. I can pare it down.

File: docs/installation/presto-admin-installation.rst (hunk 4 above)
Comment (mattsfuller) [inline docs/installation/presto-admin-installation.rst]: ok. I think this may have come from John. But it does seem unnecessary.

File: docs/installation/presto-admin-installation.rst [hunk 6]
```diff
@@ -5,19 +5,42 @@ Installing Presto Admin
...
+
+Presto Admin is packaged as an offline installer --
+``prestoadmin-<version>-offline.tar.bz2`` -- and as an online
+installer -- ``prestoadmin-<version>-online.tar.bz2``. The offline
+installer includes all of the dependencies for ``presto-admin``, so it
+can be used on a cluster without an outside network connection. The
+offline installer is currently only supported on RedHat Linux 6.x or
+CentOS equivalent.
+
+The online installer downloads all of the dependencies when you call
```
Comment (ebd2) [inline docs/installation/presto-admin-installation.rst]: Suggest: The online installer downloads all of presto-admin's dependencies when you run

File: docs/installation/presto-admin-installation.rst [hunk 7]
```diff
@@ -5,19 +5,42 @@ Installing Presto Admin
...
+``./install-prestoadmin.sh``. You need to use the online installer
+for installation of Presto on Amazon EMR.
+
+You may also need to use the online installer if you try to install
+``presto-admin`` on an operating system other than listed above,
+because some of the binary files are dependent on the system files of
+a given operating system. Be aware, though, that there may be other
```
Comment (ebd2) [inline docs/installation/presto-admin-installation.rst]: Suggest dropping "Be aware, though, that"

File: docs/installation/presto-admin-installation.rst [hunk 8]
```diff
@@ -5,19 +5,42 @@ Installing Presto Admin
...
+operating system dependent differences beyond the installation process
+for unsupported operating system, and ``presto-admin`` may not work.
+ 
+1. Copy the installer ``prestoadmin-<version>-offline.tar.bz2`` to the
```
Comment (ebd2) [inline docs/installation/presto-admin-installation.rst]: There are a lot of notes following this step in the ordered list. Suggest s/1./Before beginning the installation process, .../

File: docs/installation/presto-admin-installation.rst (hunk 5 above)
Comment (ebd2) [inline docs/installation/presto-admin-installation.rst]: I would suggest maybe not using the phrase "security hole". How about:

The sudo setup for a non-root user must have the ability to run /bin/bash as root. For increased security, consider using a dedicated account for presto-admin.

**Comment by mattsfuller** [comment]
ac

File: docs/installation/presto-admin-installation.rst [hunk 9]
```diff
@@ -5,20 +5,34 @@ Installing Presto Admin
...
+``presto-admin``, so it can be used on a cluster without an outside
+network connection. The offline installer is currently only supported
+on RedHat Linux 6.x or CentOS equivalent.
+
+The online installer downloads all of the dependencies when you run
+``./install-prestoadmin.sh``. You must use the online installer for
+installation of Presto on Amazon EMR and for use on any operating
+system not listed above. If you are using presto-admin on an
+unsupported operating system, there may be operating system
+dependencies beyond the installation process, and presto-admin may not
```
Comment (rschlussel) [inline docs/installation/presto-admin-installation.rst]: presto-admin should be quoted in back ticks \`\`presto-admin\`\`

File: docs/installation/presto-admin-configuration.rst [hunk 10]
```diff
@@ -23,6 +23,9 @@ able to login to all of the nodes via SSH as that user (see
 :ref:`ssh-configuration-label` for details on how to set that up). The
 file should be owned by root with R/W permissions (i.e. 622).
 
+.. NOTE::
+   The sudo setup for a non-root user must have the ability to run /bin/bash as root. This can be a security issue. The IT organization should take the necessary steps to address this security hole and select an appropriate presto-admin user.
```
Comment (rschlussel) [inline docs/installation/presto-admin-configuration.rst]: \`\`/bin/bash\`\`

**Comment by rschlussel** [comment]
looks good % a couple formatting things
